src/plugins/intel_cpu/src/infer_request.cpp

```diff
@@ -342,600 +354,350 @@ void InferRequestBase::changeDefaultPtr() {
+                           name,
+                           ", because model output size = ",
+                           ov::shape_size(shape.to_shape()),
+                           " and blob size = ",
+                           tensor->get_size(),
+                           " are different.");
         }
 
-        const auto &desc = graph->getOutputNodeByName(name)->getParentEdgesAtPort(0)[0]->getMemory().getDesc();
-        if (!isDynamic && blobDesc == MemoryDescUtils::convertToTensorDesc(desc)) {
-            externalPtr[name] = data;
-        } else if (externalPtr.find(name) != externalPtr.end()) {
-            externalPtr.erase(name);
+        const auto& desc = graph->getOutputNodeByName(name)->getParentEdgesAtPort(0)[0]->getMemory().getDesc();
+        if (!isDynamic && tensor_desc == MemoryDescUtils::convertToTensorDesc(desc)) {
+            external_ptr[name] = tensor;
+        } else if (external_ptr.find(name) != external_ptr.end()) {
+            external_ptr.erase(name);
         }
-        _outputs[name] = data;
+
+        m_outputs[name] = tensor;
         outputControlBlocks.erase(name); // now the memory is under user's control
     }
+    ov::ISyncInferRequest::set_tensor(port, tensor);
 }
 
-void InferRequest::SetBlobsImpl(const std::string& name, const InferenceEngine::BatchedBlob::Ptr& batched_blob) {
-    _batched_inputs[name] = batched_blob;
+void SyncInferRequest::set_tensors_impl(const ov::Output<const ov::Node> port, const std::vector<ov::SoPtr<ITensor>>& tensors) {
+    for (const auto& input : get_inputs()) {
+        if (input == port) {
+            m_batched_tensors[input.get_tensor_ptr()] = tensors;
+            return;
+        }
+    }
+    OPENVINO_THROW("Cannot find port to set_tensors!");
 }
 
-InferenceEngine::Blob::Ptr InferRequest::GetBlob(const std::string& name) {
-    OV_ITT_SCOPED_TASK(itt::domains::intel_cpu, "GetBlob");
+void SyncInferRequest::init_tensor(const std::string& name) {
+    OV_ITT_SCOPED_TASK(itt::domains::intel_cpu, "init_tensor");
 
     if (!graph || !graph->IsReady())
-        IE_THROW() << "Graph is not ready!";
+        OPENVINO_THROW("Graph is not ready!");
 
-    InferenceEngine::Blob::Ptr data;
+    OPENVINO_ASSERT(!name.empty(), "Can't prepare tensor for empty name! ");
 
-    const auto &inMap = graph->inputNodesMap;
+    ov::SoPtr<ITensor> tensor;
+    const auto& inMap = graph->inputNodesMap;
     auto input = inMap.find(name);
     if (input != inMap.end()) {
-        if (_inputs.find(name) == _inputs.end()) {
-            auto inputNode = modelInputsMap.find(name);
-            if (inputNode != modelInputsMap.end()) {
-                if (!inputNode->second) {
-                    IE_THROW() << "Can't get blob with name: " << name << ", because has null pointer to input node";
-                }
+        auto input_port = m_input_ports_map.find(name);
+        if (input_port != m_input_ports_map.end()) {
```

Comment:
```suggestion
        OPENVINO_ASSERT(input_port != m_input_ports_map.end(), "Tensor with name: ", name, " exists in CPU plugin graph, but absents in network inputs");
```
Replace if/else with early throw.